lib/service/service.go

```diff
@@ -3440,6 +3451,9 @@ func (process *TeleportProcess) waitForAppDepend() {
 
 // registerTeleportReadyEvent ensures that a TeleportReadyEvent is produced
 // when all components have started.
+// Note that this function should be kept in sync with the Config.ComponentCount function so that
```

Comment:
If the mapping is 1:1, can't we just calculate the value here and store it in the process?

Reply:
Oh, I think this is a good idea. Implemented something similar in 16b4228daee0a69529d994b7458a363f4a97adb1.